### ISSUE TITLE
dictionary like in Kaku

### ISSUE THREAD

**Issue by KonstantinDjairo** [firemaples/EverTranslator#268]
i know that we could use mdict on android, but it does not work with phrases or anything that is not a word alone
if we have something like 納期に間に合わせるため  it won't work on mdict

the way kaku works with whole phrases is awesome, it allows you to click and see the meaning in dictionary, instead of a translating.
> https://imgur.com/a/a5s3CWk

sorry about my bad english btw

**Comment by uomar8** [comment]
No need to apologize for your English

That would be helpful, the main focus was on translate
But advanced dictionary is a good thing to have

I will try to work on it and see what will happen

**Comment by KonstantinDjairo** [comment]
> > i know that we could use mdict on android, but it does not work with phrases or anything that is not a word alone if we have something like 納期に間に合わせるため it won't work on mdict
> > the way [kaku](https://github.com/0xbad1d3a5/Kaku) works with whole phrases is awesome, it allows you to click and see the meaning in dictionary, instead of a translating.
> > > https://imgur.com/a/a5s3CWk
> > 
> > 
> > sorry about my bad english btw
> 
> My English is also bad sweat_smile, it is just a negotiation tool.
> 
> It's a great suggestion. +1 I also want to have the feature you mentioned.
> 
> I just moved to Japan, but I don't understand the Japanese language, so I also use this app heavily these days. It would be very helpful if I could add this feature to EverTranslator.
> 
> Let me check how Kaku works and how to implement it in this application. Thanks for the suggestion. laughing
> 
> > No need to apologize for your English
> > That would be helpful, the main focus was on translate But advanced dictionary is a good thing to have
> > I will try to work on it and see what will happen
> 
> Thanks for your help. Do you have any idea about it? thinking

ありがとうございます!!

**Comment by uomar8** [comment]
> Thanks for your help. Do you have any idea about it? 🤔

I do have one what I noticed is the word detection is really good it can detect Japanese words as good as Kaku
![Screenshot_20230425_045814](https://user-images.githubusercontent.com/7046347/234156807-6e82a971-7b9d-48d8-b0b6-5c4915bfc01b.jpg)
![Screenshot_20230425_045834](https://user-images.githubusercontent.com/7046347/234157388-72b713fb-a461-4299-86e2-9407c6af0704.jpg)


The only thing Kaku can achieve is clicking inside the word
![Screenshot_20230425_045847](https://user-images.githubusercontent.com/7046347/234157023-745c4762-705a-40e9-97c6-340786731fd5.jpg)

And how we could achieve that is to recheck the words but remove what's before the clicked letter 
![Screenshot_20230425_050630_com example android notepad_edit_5048374310687](https://user-images.githubusercontent.com/7046347/234157540-37792144-9838-47a9-808d-058076761be4.jpg)


And in the settings we could choose between dictionary and web
Or button to switch between them

**Comment by KonstantinDjairo** [comment]
> > Thanks for your help. Do you have any idea about it? thinking
> 
> I do have one what I noticed is the word detection is really good it can detect Japanese words as good as Kaku ![Screenshot_20230425_045814](https://user-images.githubusercontent.com/7046347/234156807-6e82a971-7b9d-48d8-b0b6-5c4915bfc01b.jpg) ![Screenshot_20230425_045834](https://user-images.githubusercontent.com/7046347/234157388-72b713fb-a461-4299-86e2-9407c6af0704.jpg)
> 
> The only thing Kaku can achieve is clicking inside the word ![Screenshot_20230425_045847](https://user-images.githubusercontent.com/7046347/234157023-745c4762-705a-40e9-97c6-340786731fd5.jpg)
> 
> And how we could achieve that is to recheck the words but remove what's before the clicked letter ![Screenshot_20230425_050630_com example android notepad_edit_5048374310687](https://user-images.githubusercontent.com/7046347/234157540-37792144-9838-47a9-808d-058076761be4.jpg)
> 
> And in the settings we could choose between dictionary and web Or button to switch between them

you could add jmdict, so it would be a built-in dictionary, like in kaku, this is what make kaku so fast to use.
i noticed that they put [jmdict](https://en.wikipedia.org/wiki/JMdict) and [kanjidict2](http://www.edrdg.org/wiki/index.php/KANJIDIC_Project) in [that location](https://github.com/0xbad1d3a5/Kaku/tree/9c0d26d261f417cc86f46f56ac87f9ac73e3198e/app/src/main/java/ca/fuwafuwa/kaku/XmlParsers)

to implement a JMDict viewer they are parsing the JMDict XML file, creating a local database to store the dictionary entries, and building the UI to display the entries. Fortunately the code is already there and kaku is also writen in kotlin

**Comment by KonstantinDjairo** [comment]
Yes, but a lot of dictionaries uses the same format as jmdict, also, opening the browser for searching a word is rather slow than just displaying the respective dictionary entry.
Pretty any dictionary can be found in the freemdict website,not only japanese.
